node_exporter_test.go

```diff
@@ -0,0 +1,113 @@
+package main
+
+import (
+	"fmt"
+	"net/http"
+	"os"
+	"os/exec"
+	"testing"
+
+	"time"
+
+	"github.com/prometheus/procfs"
+)
+
+func TestFileDescriptorLeak(t *testing.T) {
+	var (
+		binary  = "./node_exporter"
+		address = "localhost:9100"
+	)
+
+	if err := checkBinaryAvailable(binary); err != nil {
+		t.Skipf("node_exporter binary not available, try to run `make build` first: %s", err)
+	}
+	if err := checkProcAvailable(); err != nil {
+		t.Skipf("proc filesystem is not available, but currently required to read number of open file descriptors: %s", err)
+	}
+
+	errc := make(chan error)
+	exporter := exec.Command(binary, "-web.listen-address", address)
+	go func() {
+		if err := exporter.Run(); err != nil {
+			errc <- fmt.Errorf("execution of node_exporter failed: %s", err)
+		} else {
+			errc <- nil
+		}
+	}()
+
+	select {
+	case err := <-errc:
+		t.Fatal(err)
+	case <-time.After(100 * time.Millisecond):
```

Comment:
This is a blocker to let the node_exporter start. Once this timeout fires, we'll continue with the testing. I briefly tried using a loop check instead of a sleep, but it became quite a bit more complicated.